### PR TITLE
survival analysis tests for the select_*() functions

### PR DESCRIPTION
**Pull request by topepo**
unit tests for #118

### CODE REVIEW

**Review by hfrick — APPROVED** [review]
The tests look good 👍 

I've one comment about the wording, let me know what you think and I can open an issue for that. 

I also left three comments about the call shown in errors and warnings. That's not survival-specific, this is something we want to do in general and have started doing bit by bit across all packages. We generally try to get it right when adding new things, however, the `select_*()` functions are not new per se. Do we want to tackle that now or as part of a general error-and-warning push for tune?

File: tests/testthat/_snaps/survival-tune-select.md [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+          <dbl> <chr>                
+      1    0.01 Preprocessor1_Model01
+
+---
+
+    Code
+      select_best(grid_dynamic_res, metric = "brier_survival", eval_time = c(5, 10))
+    Condition
+      Warning:
+      2 evaluation times are available; the first (5) will be used.
```
[the remark this answers is not in the document]
Comment (topepo) [reply tests/testthat/_snaps/survival-tune-select.md:155]: How does this sound? https://github.com/tidymodels/tune/commit/8e5f56b6b4c33f36bf3506bd9cbe3e1d5812b882

```
Warning message:
8 evaluation times are available; the first will be used (i.e. `eval_time = 0.058824`). 
```

(some random data I mocked up)

File: tests/testthat/_snaps/survival-tune-select.md [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    Condition
+      Error in `select_best()`:
+      ! "brier_survival_integrated" was not in the metric set. Please choose from: "concordance_survival".
+
+---
+
+    Code
+      select_best(grid_static_res, metric = "concordance_survival", eval_time = 0)
+    Condition
+      Warning in `show_best()`:
```
[the remark this answers is not in the document]
Comment (topepo) [reply tests/testthat/_snaps/survival-tune-select.md:37]: The call stack is

```
select_best.tune_results()
└── show_best.tune_results()
    └── choose_eval_time()
```

`choose_eval_time()` has a `call` argument but `show_best()` does not. I think that we'd have to add one to `show_best()` to get warnings/errors/msgs to register that the user's call was to  `select_best()` but please verify.

[the remark this answers is not in the document]
Comment (hfrick) [reply tests/testthat/_snaps/survival-tune-select.md:37]: Logged at https://github.com/tidymodels/tune/issues/819

File: tests/testthat/_snaps/survival-tune-select.md [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+      1    0.01 Preprocessor1_Model01
+
+# grid tuning survival models with dynamic metric
+
+    Code
+      select_best(grid_dynamic_res)
+    Condition
+      Warning in `select_best()`:
+      No value of `metric` was given; "brier_survival" will be used.
+      Warning:
```
[the remark this answers is not in the document]
Comment (hfrick) [reply tests/testthat/_snaps/survival-tune-select.md:131]: Logged at https://github.com/tidymodels/tune/issues/820

File: tests/testthat/_snaps/survival-tune-select.md [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+      Error in `select_best()`:
+      ! "brier_survival_integrated" was not in the metric set. Please choose from: "brier_survival".
+
+---
+
+    Code
+      select_best(grid_dynamic_res, metric = "brier_survival", eval_time = 0)
+    Condition
+      Error in `show_best()`:
+      ! Evaluation time 0 is not in the results.
```
[the remark this answers is not in the document]
Comment (hfrick) [reply tests/testthat/_snaps/survival-tune-select.md:176]: also at https://github.com/tidymodels/tune/issues/819

**Review by hfrick — APPROVED** [review]
I've open issues for the passing the call so we can move on with this PR :)

**Pull request merged**